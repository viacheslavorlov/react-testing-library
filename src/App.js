import './App.css';
import {useEffect, useState} from "react";

function App() {
	const [data, setData] = useState(null);
	const [toggle, setToggle] = useState(false);

	const toggleClick = () => setToggle(prevState => !prevState)

	useEffect(() => {
		setTimeout( () => {
			setData({})
		}, 100)
	}, []);

	return (
		<div className="App">
			{toggle && <div  data-testid={"toggle-element"}>toggle</div>}
			{data && <div style={toggle ? {color: 'red'} : {color: "blueviolet"}}>data</div>}
            <h1>Hello world</h1>
            <input type="text" placeholder={"...input text..."}/>
            <br/>
            <button data-testid={"toggle-btn"} onClick={toggleClick}>Click!</button>
		</div>
	);
}

export default App;
