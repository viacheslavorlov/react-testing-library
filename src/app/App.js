import './App.css';
import Users from "../users/Users";
import ToggleComponent from "../toggle-component/ToogleComponent";
import {BrowserRouter, NavLink, Route, Routes} from "react-router-dom";
import MainPage from "../pages/MainPage";
import AboutPage from "../pages/AboutPage";
import ErrorPage from "../pages/ErrorPage";
import SingleUser from "../users/SingleUser";
import AppRouter from "../routes/AppRouter";
import Navbar from "./navbar/Navbar";

function App() {


	return (
		<div className="App">
			<Navbar/>
			<AppRouter/>
		</div>
	);
}

export default App;
