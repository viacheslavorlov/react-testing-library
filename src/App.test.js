import {fireEvent, render, screen} from '@testing-library/react';
import App from './App';


describe('test app', function () {
	test('renders learn react link', () => {
		render(<App/>);
		const HelloWorld = screen.getByText(/Hello world/i);
		expect(HelloWorld).toBeInTheDocument();
		const inputElement = screen.getByPlaceholderText(/input text/);
		expect(inputElement).toBeInTheDocument();
		const button = screen.getByRole('button')
		expect(button).toBeInTheDocument();
		expect(button).toMatchSnapshot()
		expect(inputElement).toMatchSnapshot()
		screen.debug()

	});

	test('render with query', async () => {
		render(<App/>);
		const helloworldFalse = screen.queryByText(/hello2/i)
		expect(helloworldFalse).toBeNull()
		const data = await screen.findByText(/data/i)
		expect(data).toBeInTheDocument()
		expect(data).toHaveStyle({color: 'blueviolet'})
		screen.debug()
	});
	test('render with toggle &test-id', async () => {
		render(<App/>);
		const btn = screen.getByTestId('toggle-btn')
		const toggle = screen.queryByTestId('toggle-element')
		expect(btn).toBeInTheDocument()
		expect(toggle).toBeNull()
		fireEvent.click(btn)
		const toggleAfterClick = screen.queryByTestId('toggle-element')
		expect(toggleAfterClick).toBeInTheDocument()
		fireEvent.click(btn)
		const toggleAfter2Clicks = screen.queryByTestId('toggle-element')
		expect(toggleAfter2Clicks).toBeNull()
	});
});



