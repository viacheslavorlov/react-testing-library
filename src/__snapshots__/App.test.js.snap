// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test app renders learn react link 1`] = `
<button
  data-testid="toggle-btn"
>
  Click!
</button>
`;

exports[`test app renders learn react link 2`] = `
<input
  placeholder="...input text..."
  type="text"
/>
`;
